app/domain/authentication/authn_azure/application_identity.rb

```diff
@@ -0,0 +1,132 @@
+module Authentication
+  module AuthnAzure
+
+    Log = LogMessages::Authentication
+    Err = Errors::Authentication::AuthnAzure
+    # Possible Errors Raised: ConstraintNotSupported, MissingAnnotationConstraint, IllegalConstraintCombinations
+
+    # This class defines an application identity of a given Conjur host.
+    # The constructor initializes an ApplicationIdentity object and validates that
+    # it is configured correctly.
+    # The difference between the validation in this constructor and
+    # the validation in ValidateApplicationIdentity is that here we validate that
+    # the application identity is configured correctly, and thus is a valid application
+    # identity. In ValidateApplicationIdentity we validate that the defined application
+    # identity is actually the one we expect from Azure. This validation is done through
+    # a combination of comparisons that are performed based on the type of assigned
+    # identity provided.
+    # For example, an application identity `some-namespace/blah/some-value` is not
+    # valid and will fail the validation here. However, an application identity
+    # `some-namespace/service-account/some-value` is a valid application identity
```

Comment:
same here. Please also give context to _why_ it is a valid application and may fail in the `validate_application_identity` validations. It will help to understand the differences between the 2 validations

Reply:
After defining and giving examples of invalid and valid application identities, adding explanation that 
"`authn-azure/<service-id>/subscription-id` is a valid application identity because `subscription_id` is one of the defined, accepted constraints and will pass the validation here."

Reply:
The example `some-namespace/service-account/some-value` refers to when the app id is in the host id and not in annotations. Please change your examples to hosts with app id defined in annotations